src/PostgREST/Private/QueryFragment.hs

```diff
@@ -90,11 +90,13 @@ asCsvF = asCsvHeaderF <> " || '\n' || " <> asCsvBodyF
       ")"
     asCsvBodyF = "coalesce(string_agg(substring(_postgrest_t::text, 2, length(_postgrest_t::text) - 2), '\n'), '')"
 
-asJsonF :: SqlFragment
-asJsonF = "coalesce(json_agg(_postgrest_t), '[]')::character varying"
+asJsonF :: Bool -> SqlFragment
+asJsonF False = "coalesce(json_agg(_postgrest_t), '[]')::character varying"
+asJsonF True  = "coalesce(json_agg(_postgrest_t.pgrst_scalar), '[]')::character varying"
```

Comment:
The Bool value here could use some clarification. How about changing the function to:

```suggestion
asJsonF returnsScalar = if returnsScalar
  then "coalesce(json_agg(_postgrest_t.pgrst_scalar), '[]')::character varying"
  else "coalesce(json_agg(_postgrest_t), '[]')::character varying"
```

Reply:
It could also be like:

```suggestion
asJsonF returnsScalar | returnsScalar = "coalesce(json_agg(_postgrest_t.pgrst_scalar), '[]')::character varying"
                      | otherwise     = "coalesce(json_agg(_postgrest_t), '[]')::character varying"
```

Reply:
So now you're doing multi-line suggestions... cool :D

I went with the 2nd suggestion, but moved it to a new line. Will push and merge in a sec.